core/src/main/scala/akka/kafka/javadsl/PartitionAssignmentHandler.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.javadsl
+
+import akka.kafka.RestrictedConsumer
+import akka.annotation.ApiMayChange
+import org.apache.kafka.common.TopicPartition
+
+/**
+ * The API is new and may change in further releases.
+ *
+ * Allows to execute user code when Kafka rebalances partitions between consumers, or an Alpakka Kafka consumer is stopped.
+ * Use with care: These callbacks are called synchronously on the same thread Kafka's `poll()` is called.
+ * A warning will be logged if a callback takes longer than the configured `partition-handler-warning`.
+ *
+ * There is no point in calling `CommittableOffset`'s commit methods as their committing won't be executed as long as any of
+ * the callbacks in this class are called.
+ *
+ * This complements the methods of Kafka's [[org.apache.kafka.clients.consumer.ConsumerRebalanceListener ConsumerRebalanceListener]] with
+ * an `onStop` callback.
```

Comment:
To match the scaladsl comments.

```suggestion
 * an `onStop` callback which is called before `Consumer.close`.
```